src/components/TabPanel/index.tsx

```diff
@@ -50,22 +50,30 @@ export interface TabPanelProps {
     /**
      * Current index of tab
      */
-    currentTabIndex: number;
+    currentTabIndex?: number;
     /**
      * Optinal JSX element to head
      */
     optionalHead?: React.ReactNode;
 }
 
 
+interface TabPanelState {
+    currentTab: number;
+}
+
 /**
  * Component for switching between different tabs on one page.
  */
-export class TabPanel extends React.Component<TabPanelProps> {
+export class TabPanel extends React.Component<TabPanelProps, TabPanelState> {
     public static defaultProps = {
         hideMode: HideMode.hide,
     };
 
+    public state = {
+        currentTab: 0,
+    };
```

Comment:
why did you add one more index if this component already has prop currentTabIndex?